internal/cmd/managedcmd.go

```diff
@@ -80,14 +81,14 @@ func (c *Config) runManagedCmd(cmd *cobra.Command, args []string, sourceState *c
 			}
 
 			var path fmt.Stringer
-			switch c.managed.pathStyle {
-			case chezmoi.PathStyleAbsolute:
+			switch c.managed.pathStyle.String() {
+			case "absolute":
```

Comment:
This basically means there are no more compile time checks.

How about this?

https://go.dev/play/p/DEry6kT1yBX

Reply:
I don't understand how `Optional[T]` would help here. Note that Go has basically zero support for enums, even though I wish it had.

Reply:
The main idea there was storing actual typed values inside choiceFlag instead of strings. But then I found that Go doesn't check types and will accept any string for `choiceFlag[PathStyle]` value.

I'd still prefer using fixed constants to strings though. This makes it easier to understand what values can used and protects from typos.